src/atom_utils.F

```diff
@@ -1438,32 +1455,96 @@ SUBROUTINE potential_numeric(cpot, density, nu, grid)
          int2 = int2 + density(i)*wr(i)/r(i)**(nu + 1)
       END DO
 
-   END SUBROUTINE potential_numeric
+   END SUBROUTINE potential_coulomb_numeric
+
+! **************************************************************************************************
+!> \brief ...
+!> \param cpot ...
+!> \param density ...
+!> \param nu ...
+!> \param grid ...
+!> \param omega ...
+!> \param kernel ...
+!> \param do_gh ...
+! **************************************************************************************************
+   SUBROUTINE potential_longrange_numeric(cpot, density, nu, grid, omega, kernel, do_gh)
```

Comment:
This function is actually two functions: you could move the branch up the callstack and split this one instead.

Reply:
Done.